packages/ngtools/webpack/src/paths-plugin.ts

```diff
@@ -154,40 +158,75 @@ export class TypeScriptPathsPlugin {
         // For example, if the first one resolves, any others are not needed and do not need
         // to be created.
         const replacements = findReplacements(originalRequest, this.patterns);
+        const basePath = this.baseUrl ?? '';
+
+        const attemptResolveRequest = (request: DoResolveValue): Promise<DoResolveValue | null> => {
+          return new Promise((resolve, reject) => {
+            resolver.doResolve(
+              target,
+              request,
+              '',
+              resolveContext,
+              (error: Error | null, result: DoResolveValue) => {
+                if (error) {
+                  reject(error);
+                } else if (result) {
+                  resolve(result);
+                } else {
+                  resolve(null);
+                }
+              },
+            );
+          });
+        };
 
-        const tryResolve = () => {
+        const tryNextReplacement = () => {
           const next = replacements.next();
           if (next.done) {
             callback();
 
             return;
           }
 
-          const potentialRequest = {
+          const targetPath = path.resolve(basePath, next.value);
+          // If there is no extension. i.e. the target does not refer to an explicit
+          // file, then this is a candidate for module/package resolution.
+          const canBeModule = path.extname(targetPath) === '';
+
+          // Resolution in the target location, preserving the original request.
+          // This will work with the `resolve-in-package` resolution hook, supporting
+          // package exports for e.g. locally-built APF libraries.
+          const potentialRequestAsPackage = {
             ...request,
-            request: path.resolve(this.baseUrl ?? '', next.value),
+            path: targetPath,
             typescriptPathMapped: true,
           };
 
-          resolver.doResolve(
-            target,
-            potentialRequest,
-            '',
-            resolveContext,
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            (error: Error | null, result: any) => {
-              if (error) {
-                callback(error);
-              } else if (result) {
-                callback(undefined, result);
-              } else {
-                tryResolve();
-              }
-            },
-          );
+          // Resolution in the original callee location, but with the updated request
+          // to point to the mapped target location.
+          const potentialRequestAsFile = {
+            ...request,
+            request: targetPath,
+            typescriptPathMapped: true,
+          };
```

Comment:
nit: Could probably reduce the amount of code needed by creating another generator function (`createReplacementRequests` ?) that produces the request objects to try. Each replacement would either yield one or two values depending on the extension check.
Then this could be setup to be `const requests = createReplacementRequests(replacements);` and then `requests` would be used instead of `replacements`.

Reply:
Given you declaring this as a `nit`, I assume we can clean this up in a follow-up. I don't want to make significant further changes to this PR which already took up a surprising amount of time (due to exploring various approaches, reversing to understand enhanced-resolve etc.)